doc/source/notebooks/basics/classification-redesigned.pct.py

```diff
@@ -0,0 +1,207 @@
+
+# %%
+# Select some input locations
+ind = rng.randint(0, 200, (30,))
+X_gen = x_grid[ind]
+
+# evaluate probability and get Bernoulli draws
+p = p_grid[ind, 1:2]
+Y_gen = rng.binomial(1, p)
+
+# plot
+plt.plot(x_grid, p_grid[:, 1], "C1", linewidth=2)
+plt.plot(X_gen, p, "C1o", ms=6)
+_ = plt.plot(X_gen, Y_gen, "C3x", ms=8, mew=2)
+
+# %% [markdown]
+# ### Implementation with GPflow
+#
+# For the model described above, the posterior $f(x)|Y$ (say $p$) is not Gaussian any more and does not have a closed-form expression.
+# A common approach is then to look for the best approximation of this posterior by a tractable distribution (say $q$) such as a Gaussian distribution.
+# In variational inference, the quality of an approximation is measured by the Kullback-Leibler divergence $\mathrm{KL}[q \| p]$.
+# For more details on this model, see Nickisch and Rasmussen (2008).
+#
+# The inference problem is thus turned into an optimization problem: finding the best parameters for $q$.
+# In our case, we introduce $U \sim \mathcal{N}(q_\mu, q_\Sigma)$, and we choose $q$ to have the same distribution as $f | f(X) = U$.
+# The parameters $q_\mu$ and $q_\Sigma$ can be seen as parameters of $q$, which can be optimized in order to minimise  $\mathrm{KL}[q \| p]$.
+#
+# This variational inference model is called `VGP` in GPflow:
+
+# %%
+m = gpflow.models.VGP(
+    (X, Y), likelihood=gpflow.likelihoods.Bernoulli(), kernel=gpflow.kernels.Matern52()
+)
+gpflow.set_trainable(m.kernel, False)
+opt = gpflow.optimizers.Scipy()
+opt.minimize(m.training_loss, variables=m.trainable_variables)
+
+# %% [markdown]
+# We can now inspect the result of the optimization with `gpflow.utilities.print_summary(m)`:
+
+# %%
+gpflow.utilities.print_summary(m, fmt="notebook")
+
+# %% [markdown]
+# In this table, the first two lines are associated with the kernel parameters, and the last two correspond to the variational parameters.
+# **NOTE:** In practice, $q_\Sigma$ is actually parameterized by its lower-triangular square root $q_\Sigma = q_\text{sqrt} q_\text{sqrt}^T$ in order to ensure its positive-definiteness.
+#
+# For more details on how to handle models in GPflow (getting and setting parameters, fixing some of them during optimization, using priors, and so on), see [Manipulating GPflow models](../understanding/models.ipynb).
+
+# %% [markdown]
+# ### Predictions
+#
+# Finally, we will see how to use model predictions to plot the resulting model.
+# We will replicate the figures of the generative model above, but using the approximate posterior distribution given by the model.
+
+# %% [markdown]
+# ## The conditional output distribution
+#
+# Here we show how to get the conditional output distribution and plot samples from it. In order to plot the uncertainty associated with that, we also get the percentiles from a sample of the corresponding likelihood parameter values, because the those of the output values are binary and would neither be convenient nor interesting to plot.
```

Comment:
I'm not sure what is meant with the second bit of the sentence: 

> , because the those of the output values are binary and would neither be convenient nor interesting to plot